test/e2e/network/dns_common.go

```diff
@@ -613,3 +613,35 @@ func generateDNSUtilsPod() *v1.Pod {
 		},
 	}
 }
+
+func getDNSDomain(f *framework.Framework) string {
+
+	createUtilsPod := generateDNSUtilsPod()
+	createUtilsPod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(createUtilsPod)
+	Expect(err).NotTo(HaveOccurred(), "failed to create pod: %s", createUtilsPod.Name)
+	framework.Logf("Created pod %v", createUtilsPod)
+	defer func() {
+		framework.Logf("Deleting pod %s...", createUtilsPod.Name)
+		if err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Delete(createUtilsPod.Name, metav1.NewDeleteOptions(0)); err != nil {
+			framework.Failf("Failed to delete pod %s: %v", createUtilsPod.Name, err)
+		}
+	}()
+	Expect(f.WaitForPodRunning(createUtilsPod.Name)).NotTo(HaveOccurred(), "failed to wait for pod %s to be running", createUtilsPod.Name)
+	cmd := []string{"cat", "/etc/resolv.conf"}
+	stdout, stderr, err := f.ExecWithOptions(framework.ExecOptions{
+		Command:       cmd,
+		Namespace:     f.Namespace.Name,
+		PodName:       createUtilsPod.Name,
+		ContainerName: "util",
+		CaptureStdout: true,
+		CaptureStderr: true,
+	})
+	Expect(err).NotTo(HaveOccurred(), "failed to examine resolv.conf file on pod, stdout: %v, stderr: %v, err: %v", stdout, stderr, err)
+	res := strings.Split(stdout, "\n")
+	dnsDomain := strings.Split(res[1], " ")
+
+	if !strings.Contains(dnsDomain[1], "default.svc") && !strings.Contains(dnsDomain[1], dnsDomain[3]) {
```

Comment:
I think you just need to verify first that your dnsDomain array has 4 elements at least ...

```
if len(gnsDomain)<4 || (!strings.Contains(dnsDomain[1], "default.svc") && !strings.Contains(dnsDomain[1], dnsDomain[3])) {
   then fail
}
```